src/relay/backend/build_module.cc

```diff
@@ -59,17 +64,50 @@ struct BuildOutput {
  */
 struct GraphCodegen {
  public:
-  GraphCodegen() {
-    auto pf = GetPackedFunc("relay.build_module._GraphExecutorCodegen");
-    mod = (*pf)();
+  explicit GraphCodegen(Target target_host) : target_host_(target_host) {
+    const String executor_str =
+        target_host->GetAttr<String>("executor").value_or(kTvmExecutorGraph);
+    if (executor_str == kTvmExecutorGraph) {
+      executor_ = ExecutorType::Graph;
+      auto pf = GetPackedFunc("relay.build_module._GraphExecutorCodegen");
+      mod = (*pf)();
+    } else if (executor_str == kTvmExecutorAot) {
```

Comment:
I think we'd need a ExecutorCodegen bass class and specialization of GraphCodegen and AOTCodegen. The switch between the which functionality to run should be done in a higher-level in the build_module.cc. Therefore we can avoid the having to check which executor inside each of these functions. WDYT ?